### PR TITLE
Read instance category on update check

### PR DESCRIPTION
**Pull request by Altahrim**
Allows to read new attributes `category` and `isSubscriber` from update check

### CODE REVIEW

**Review by artonge — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Review by skjnldsv — COMMENTED** [review]
I'd like @blizzz approval before merge please

File: src/Request.php [hunk 1]
```diff
@@ -145,34 +163,13 @@ public function getPHPReleaseVersion() {
...
-			// starting with nextcloud 17 there is no production channel anymore
-			if ($this->channel === 'production') {
-				$this->channel = 'stable';
-			}
-
-			if(count($version) === 12) {
+		$this->remoteAddress = (isset($server['REMOTE_ADDR']) ? $server['REMOTE_ADDR'] : '');
+		switch (count($version)) {
+			case 14:
+				$this->isSubscriber = (int) $version[13];
```
Comment (blizzz) [inline src/Request.php:169]: does it mean we rely on the client to provide reliable subscription information?

**Review by blizzz — APPROVED** [review]
(just the other question)

File: src/Request.php (hunk 1 above)
Comment (Altahrim) [reply src/Request.php:169]: Yes,I implemented it here : https://github.com/nextcloud/server/pull/39264
I didn't find any other way, but we can discuss it if you have some time :)

**Pull request merged**